### ISSUE TITLE
Use TensorFlow dataset object for training data handling

### ISSUE THREAD

**Issue by generic-github-user** [generic-github-user/icon-encoder#14]
As described in [this article](https://towardsdatascience.com/how-to-use-dataset-in-tensorflow-c758ef9e4428). In theory, [datasets](https://www.tensorflow.org/guide/datasets) are more efficient when training. Additionally, old methods such as [`tf.train.shuffle_batch`](https://www.tensorflow.org/api_docs/python/tf/train/shuffle_batch) are depreciated and will be removed in the future. It is in our best interest to keep the code as modern and up-to-date as possible.

**Comment by generic-github-user** [comment]
This allows us to easily batch the training data and improve performance and reliability (and so on).
